User-tools: Add qualfication tool support for Onprem platform (#284)
NVIDIA/spark-rapids-tools: user_tools/docs/user-tools-onprem.md

# RAPIDS User Tools on OnPrem platform

> This is a guide for the RAPIDS tools for Apache Spark on OnPrem platform. At the end of this guide, the user will be able to run the RAPIDS tools to analyze the clusters and the applications running OnPrem.

Reviewer: I am not sure it is accurate to say `OnPrem platform`.
Maybe we should rephrase t o elaborate what we mean by OnPrem just in case someone can confuse it that it is some sort of a cloud product. for example
```suggestion
This is a guide for the RAPIDS tools for Apache Spark clusters that are provisioned manually (a.k.a onPrem). At the end of this guide, the user will be able to run the RAPIDS tools to analyze the clusters and the applications running OnPrem.
```

2. Internet access to download JAR dependencies from mvn: `spark-*.jar`
3. Dependencies are cached on the local disk to reduce the overhead of the download.

#### Command options

| Option                         | Description                                                                                                                                                                                                                                                                                               | Default                                                                                                                                              | Required |
|--------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------|:--------:|

> | **cpu_cluster**                | The Onprem-cluster on which the Apache Spark applications were executed. Accepted values are an OnPrem-cluster name, or a valid path to the cluster properties file (json format)                                                                                                                         | N/A                                                                                                                                                  |     Y    |

Reviewer: should be a valid path only.

Author: Done.

| **rapids_options****           | A list of valid [Qualification tool options](../../core/docs/spark-qualification-tool.md#qualification-tool-options). Note that (`output-directory`, `platform`) flags are ignored, and that multiple "spark-property" is not supported.                                                                  | N/A                                                                                                                                                  |     N    |

#### Use case scenario

A typical workflow to successfully run the `qualification` command in local mode is described as follows:

1. Store the Apache Spark event logs in local directory.
2. User sets up the development machine:

>    1. configures Java

Reviewer: I know that this is the same line from other platforms. I just thought that the instruction is too vague.
We want to give some specific about what "configures java" mean. Or we replace it with something similar to  `on a machine with Java SDK 1.8+ installed1.

   2. installs `spark_rapids_user_tools`
3. User defines the onprem-cluster on which the Spark application were running. Note that the cluster does not have to be
   active.
4. The following script runs qualification tool locally:

   ```
   # define the wrapper cache directory if necessary
   export RAPIDS_USER_TOOLS_CACHE_FOLDER=my_cache_folder

>    export EVENTLOGS=/LOGS_BUCKET/eventlogs/

Reviewer: `LOGS_BUCKET`  ?

Author: I hadn't specified LOGS_BUCKET but it was meant to be path_to_eventlogs. Updated it to reflect the same

   export CLUSTER_NAME=my-onprem-cpu-cluster
   
   spark_rapids_user_tools onprem qualification \
      --eventlogs $EVENTLOGS \

>       --cpu_cluster $CLUSTER_NAME

Reviewer: We cannot use `CLUSTER_NAME` for on-prem. We need a cluster property file. right?

Author: You are correct. Initially, I thought it could be either of them but it will be a cluster property file for now. Updated it.
